help.go

```diff
@@ -352,27 +363,32 @@ func printHelpCustom(out io.Writer, templ string, data interface{}, customFuncs
 
 	w := tabwriter.NewWriter(out, 1, 8, 2, ' ', 0)
 	t := template.Must(template.New("help").Funcs(funcMap).Parse(templ))
-	t.New("helpNameTemplate").Parse(helpNameTemplate)
-	t.New("usageTemplate").Parse(usageTemplate)
-	t.New("descriptionTemplate").Parse(descriptionTemplate)
-	t.New("visibleCommandTemplate").Parse(visibleCommandTemplate)
-	t.New("copyrightTemplate").Parse(copyrightTemplate)
-	t.New("versionTemplate").Parse(versionTemplate)
-	t.New("visibleFlagCategoryTemplate").Parse(visibleFlagCategoryTemplate)
-	t.New("visibleFlagTemplate").Parse(visibleFlagTemplate)
-	t.New("visibleGlobalFlagCategoryTemplate").Parse(strings.Replace(visibleFlagCategoryTemplate, "OPTIONS", "GLOBAL OPTIONS", -1))
-	t.New("authorsTemplate").Parse(authorsTemplate)
-	t.New("visibleCommandCategoryTemplate").Parse(visibleCommandCategoryTemplate)
-
-	err := t.Execute(w, data)
-	if err != nil {
-		// If the writer is closed, t.Execute will fail, and there's nothing
-		// we can do to recover.
-		if os.Getenv("CLI_TEMPLATE_ERROR_DEBUG") != "" {
-			_, _ = fmt.Fprintf(ErrWriter, "CLI TEMPLATE ERROR: %#v\n", err)
-		}
-		return
-	}
+	_, err := t.New("helpNameTemplate").Parse(helpNameTemplate)
+	handleTemplateError(err)
+	_, err = t.New("usageTemplate").Parse(usageTemplate)
+	handleTemplateError(err)
+	_, err = t.New("descriptionTemplate").Parse(descriptionTemplate)
+	handleTemplateError(err)
+	_, err = t.New("visibleCommandTemplate").Parse(visibleCommandTemplate)
+	handleTemplateError(err)
+	_, err = t.New("copyrightTemplate").Parse(copyrightTemplate)
+	handleTemplateError(err)
+	_, err = t.New("versionTemplate").Parse(versionTemplate)
+	handleTemplateError(err)
+	_, err = t.New("visibleFlagCategoryTemplate").Parse(visibleFlagCategoryTemplate)
+	handleTemplateError(err)
+	_, err = t.New("visibleFlagTemplate").Parse(visibleFlagTemplate)
+	handleTemplateError(err)
+	_, err = t.New("visibleGlobalFlagCategoryTemplate").Parse(strings.Replace(visibleFlagCategoryTemplate, "OPTIONS", "GLOBAL OPTIONS", -1))
+	handleTemplateError(err)
+	_, err = t.New("authorsTemplate").Parse(authorsTemplate)
+	handleTemplateError(err)
+	_, err = t.New("visibleCommandCategoryTemplate").Parse(visibleCommandCategoryTemplate)
+	handleTemplateError(err)
+
+	err = t.Execute(w, data)
+	handleTemplateError(err)
```

Comment:
imho this is a great place to use the "iferr" form so that the `err` var isn't getting overwritten and adding to the potential for unexpected debug output, and the last case can be folded up without assigning `err` at all 👍🏼